deps/src/caller.cpp

```diff
@@ -0,0 +1,340 @@
+#include "caller.h"
+
+#include <Singular/tok.h>
+#include <Singular/grammar.h>
+#include <Singular/ipshell.h>
+#include <Singular/lists.h>
+#include <misc/intvec.h>
+
+// #include <julia/julia.h>
+
+static jl_value_t * jl_int64_vector_type;
+static jl_value_t * jl_int64_matrix_type;
+static jl_value_t * jl_singular_number_type;
+static jl_value_t * jl_singular_poly_type;
+static jl_value_t * jl_singular_ring_type;
+static jl_value_t * jl_singular_ideal_type;
+static jl_value_t * jl_singular_matrix_type;
+static jl_value_t * jl_singular_bigint_type;
+static jl_value_t * jl_singular_bigintmat_type;
+static jl_value_t * jl_singular_map_type;
+static jl_value_t * jl_singular_resolution_type;
+
+static jl_value_t * get_type_mapper()
+{
+    std::vector<std::pair<int, std::string>> types = {
+        std::pair<int, std::string>(BIGINT_CMD, "BIGINT_CMD"),
+        std::pair<int, std::string>(NUMBER_CMD, "NUMBER_CMD"),
+        std::pair<int, std::string>(RING_CMD, "RING_CMD"),
+        std::pair<int, std::string>(POLY_CMD, "POLY_CMD"),
+        std::pair<int, std::string>(IDEAL_CMD, "IDEAL_CMD"),
+        std::pair<int, std::string>(INT_CMD, "INT_CMD"),
+        std::pair<int, std::string>(STRING_CMD, "STRING_CMD"),
+        std::pair<int, std::string>(LIST_CMD, "LIST_CMD"),
+        std::pair<int, std::string>(INTMAT_CMD, "INTMAT_CMD"),
+        std::pair<int, std::string>(BIGINTMAT_CMD, "BIGINTMAT_CMD"),
+        std::pair<int, std::string>(MAP_CMD, "MAP_CMD"),
+        std::pair<int, std::string>(RESOLUTION_CMD, "RESOLUTION_CMD"),
+        std::pair<int, std::string>(INTVEC_CMD, "INTVEC_CMD")};
```

Comment:
Wow, that's horribly inefficient. Though of course you only call `get_type_mapper` once, so it doesn't matter... but have you considered just doing this instead?
```C
struct { int cmd; const char *name; } types = {
    { BIGINT_CMD, "BIGINT_CMD" },
    ...
    { INTVEC_CMD, "INTVEC_CMD" },
};
```
And then replace `types.size()` by `sizeof(types)/siezof(types[0])`, `types[i].first` by `types[i].type` and `types[i].second.c_str()` by `types[i].name`.

Reply:
I see that various types are missing, but of course that's fine since this is just a prototype anyway :-).

Reply:
It is only for the initial setup. I kinda want to replace it by something generated in the future.

Reply:
Added that (minus smaller typos in your suggestion), but disabled clang-format for it, as it would put several entries in one line.